dll/win32/shell32/CShellDispatch.cpp

```diff
@@ -281,19 +319,106 @@ HRESULT STDMETHODCALLTYPE CShellDispatch::FindPrinter(BSTR name, BSTR location,
 HRESULT STDMETHODCALLTYPE CShellDispatch::GetSystemInformation(BSTR name, VARIANT *ret)
 {
     TRACE("(%p, %ls, %p)\n", this, name, ret);
+
+    if (!lstrcmpiW(name, L"ProcessorArchitecture"))
```

Comment:
Is it guaranteed the BSTR string is always NULL-terminated?

Reply:
Yes. They may also contain NULL characters when they contain binary data but since there is no SysStrCmp everyone just ignores this loophole. You can't even create such a binary BSTR when called from a scripting language.